main/res/layout/search_activity.xml

```diff
@@ -17,68 +17,94 @@
             <View style="@style/separator_horizontal" />
             <TextView
                 style="@style/separator_horizontal_headline"
-                android:text="@string/search_coordinates" />
+                android:text="@string/search_filter" />
         </RelativeLayout>
 
-        <Button
-            android:id="@+id/buttonLatitude"
-            style="@style/button_full"
-            android:hint="@string/latitude" />
+        <RelativeLayout
+            android:layout_height="wrap_content"
+            android:layout_width="match_parent">
 
-        <Button
-            android:id="@+id/buttonLongitude"
-            style="@style/button_full"
-            android:hint="@string/longitude" />
+            <Button
+                android:id="@+id/search_filter"
+                style="@style/button_full"
+                android:text="@string/search_filter_button"
+                android:layout_alignParentLeft="true"
+                android:layout_toLeftOf="@+id/search_filter_info"
+                android:layout_centerVertical="true"
+                android:layout_width="wrap_content"/>
 
-        <Button
-            android:id="@+id/search_coordinates"
-            style="@style/button_full"
-            android:text="@string/search_coordinates_button" />
-        <!-- ** -->
+            <ImageView
+                android:id="@+id/search_filter_info"
+                android:layout_width="36dp"
+                android:layout_height="36dp"
+                app:srcCompat="@drawable/settings_info"
+                android:layout_marginLeft="0dp"
+                android:layout_marginRight="10dp"
+                android:layout_alignParentRight="true"
+                android:layout_centerVertical="true" />
+        </RelativeLayout>
 
         <RelativeLayout style="@style/separator_horizontal_layout" >
             <View style="@style/separator_horizontal" />
             <TextView
                 style="@style/separator_horizontal_headline"
-                android:text="@string/search_address" />
+                android:text="@string/search_geo" />
         </RelativeLayout>
 
         <com.google.android.material.textfield.TextInputLayout style="@style/textinput_autocompletetextview">
             <AutoCompleteTextView
-                android:id="@+id/address"
+                android:id="@+id/geocode"
                 style="@style/textinput_embedded"
-                android:hint="@string/search_address"
-                android:imeOptions="actionGo" />
+                android:hint="@string/search_geo"
+                android:imeOptions="actionGo"
+                android:inputType="textVisiblePassword"
```

Comment:
Knowing, that this inputType was in the code before, still the question: Why shall it use that inputType?

Reply:
Good question.
Originally (at least long before) it was declared as
~~~
android:inputType="textCapCharacters"
~~~
which sets all characters to uppercase.
With 7aff8e5de155af144a76663158365d02df1fb794 it was changed to 
~~~
android:inputType="textVisiblePassword"
~~~
Now after 42e195e20c8c0446887f6babe4334d6f0ccf907b we have the implementation using viewbindings and a programmatic change in capital letters
~~~
        binding.geocode.setFilters(new InputFilter[] { new InputFilter.AllCaps() });
        binding.trackable.setFilters(new InputFilter[] { new InputFilter.AllCaps() });
~~~

The result - the declaration in the xml is superfluous.